arch/risc-v/src/esp32c3/esp32c3_irq.c

```diff
@@ -211,10 +215,6 @@ void up_irqinitialize(void)
   g_cpu_intmap[ESP32C3_CPUINT_RWBLE] = CPUINT_ASSIGN(ESP32C3_IRQ_RWBLE);
 #endif
 
-  /* Initialize CPU interrupts */
-
-  esp32c3_cpuint_initialize();
-
```

Comment:
This should not be moved above. Instead move the attribution of `g_cpu_intmap` for the special cases after `esp32c3_cpuint_initialize`, as it is done for ESP32:
https://github.com/apache/nuttx/blob/8845e6bb386e58ba1b82d68482a5a8302d3df573/arch/xtensa/src/esp32/esp32_irq.c#L452-L483

Reply:
The `esp32c3_cpuint_initialize` will initialize (memset) the g_cpu_intmap[] with UNASSIGNED, then these hardcode values attributed later are fine. So to keep it in since with ESP32 I will move it on ESP32 too.